### PR TITLE
feat: add periodic push to svc-gis

### PR DESCRIPTION
**Pull request by amsmith-pro**

### CODE REVIEW

File: server/src/config.rs [hunk 1]
```diff
@@ -31,6 +44,10 @@ impl Config {
         log::warn!("Creating Config object with default values.");
         Config {
             docker_port_grpc: 50051,
+            gis_host_grpc: String::from("svc-gis"),
```
Comment (owlot) [inline server/src/config.rs:48]: These new fields should be added to the tests below as well

File: server/src/grpc/server.rs [hunk 2]
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
...
+        .map(
+            |(label, coordinates)| svc_gis_client_grpc::client::Waypoint {
+                label: label.clone(),
+                location: Some(*coordinates),
+            },
+        )
+        .collect();
+
+    if nodes.is_empty() {
+        grpc_warn!("(waypoints_loop) no waypoints to update.");
```
Comment (owlot) [inline server/src/grpc/server.rs]: wrong function name in debug message:
(waypoints_loop) -> (update_waypoints)

File: server/src/grpc/server.rs [hunk 3]
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
...
-            self.region.get_region()
-        );
-        grpc_debug!("(request_restrictions) request: {:?}", request);
-        self.region
-            .request_restrictions(self.restrictions.clone(), request)
+    let request = tonic::Request::new(UpdateWaypointsRequest { waypoints: nodes });
+    let address = format!("{}:{}", host, port);
+    let result = GisClient::connect(address).await;
+    let Ok(mut client) = result else {
+        grpc_error!("Failed to connect to gRPC server: {}", result.unwrap_err());
```
Comment (owlot) [inline server/src/grpc/server.rs]: function name missing in debug message

File: server/src/grpc/server.rs [hunk 4]
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
...
+        return;
+    };
+
+    match client.update_waypoints(request).await {
+        Ok(response) => {
+            grpc_info!("RESPONSE={:?}", response);
```
Comment (owlot) [inline server/src/grpc/server.rs]: Should this really be an info message? Feels more like a debug message
Also add the function name prefix

File: server/src/grpc/server.rs [hunk 5]
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
...
+        }
+        Err(e) => {
+            grpc_error!("ERROR={:?}", e);
```
Comment (owlot) [inline server/src/grpc/server.rs]: Function name missing in debug message

File: server/src/grpc/server.rs [hunk 6]
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
...
+    port: u16,
+    restrictions: &HashMap<String, RestrictionDetails>,
+) {
+    let mut zones: Vec<NoFlyZone> = vec![];
+    for (label, details) in restrictions.iter() {
+        let time_start = match details.timestamp_start {
+            Some(t) => match datetime_to_timestamp(&t) {
+                Some(t) => Some(t),
+                _ => {
+                    grpc_error!("(restrictions_loop) could not convert timestamp for zone with label {label}.");
```
Comment (owlot) [inline server/src/grpc/server.rs]: Wrong function name in log message -> (update_restrictions)

File: server/src/grpc/server.rs [hunk 7]
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
...
+                }
+            },
+            None => None,
+        };
+
+        let time_end = match details.timestamp_end {
+            Some(t) => match datetime_to_timestamp(&t) {
+                Some(t) => Some(t),
+                _ => {
+                    grpc_error!("(restrictions_loop) could not convert timestamp for zone with label {label}.");
```
Comment (owlot) [inline server/src/grpc/server.rs]: Wrong function name in log message -> (update_restrictions)

File: server/src/grpc/server.rs [hunk 8]
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
...
+                    longitude: v.longitude,
+                })
+                .collect(),
+            time_start,
+            time_end,
+        });
+    }
+
+    if zones.is_empty() {
+        grpc_warn!("(restrictions_loop) no restrictions to update.");
```
Comment (owlot) [inline server/src/grpc/server.rs]: Wrong function name in log message -> (update_restrictions)

File: server/src/grpc/server.rs [hunk 9]
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
...
+        return;
+    }
+
+    let request = tonic::Request::new(UpdateNoFlyZonesRequest { zones });
+    let address = format!("{}:{}", host, port);
+    let result = GisClient::connect(address).await;
+    let Ok(mut client) = result else {
+        grpc_error!("Failed to connect to gRPC server: {}", result.unwrap_err());
```
Comment (owlot) [inline server/src/grpc/server.rs]: Missing function name in log message

File: server/src/grpc/server.rs [hunk 10]
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
...
+        return;
+    };
+
+    match client.update_no_fly_zones(request).await {
+        Ok(response) => {
+            grpc_info!("RESPONSE={:?}", response);
```
Comment (owlot) [inline server/src/grpc/server.rs]: I think this should be a debug instead again
Also missing function name prefix

File: server/src/grpc/server.rs [hunk 11]
```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
...
+        }
+        Err(e) => {
+            grpc_error!("ERROR={:?}", e);
```
Comment (owlot) [inline server/src/grpc/server.rs]: Missing function name prefix

File: server/src/grpc/server.rs [hunk 12]
```diff
@@ -357,58 +450,4 @@ mod tests {
         println!("{:?}", result);
         assert_eq!(result.released, true);
     }
-
```
Comment (owlot) [inline server/src/grpc/server.rs:360]: New tests are missing for the newly added functions

**Review by owlot — COMMENTED** [review]
I've marked mostly log message issues.
And I think some tests can be added.

File: server/src/config.rs (hunk 1 above)
Comment (amsmith-pro) [reply server/src/config.rs:48]: https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

File: server/src/grpc/server.rs (hunk 2 above)
Comment (amsmith-pro) [reply server/src/grpc/server.rs]: https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

File: server/src/grpc/server.rs (hunk 3 above)
Comment (amsmith-pro) [reply server/src/grpc/server.rs]: https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

File: server/src/grpc/server.rs (hunk 4 above)
Comment (amsmith-pro) [reply server/src/grpc/server.rs]: https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

File: server/src/grpc/server.rs (hunk 5 above)
Comment (amsmith-pro) [reply server/src/grpc/server.rs]: https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

File: server/src/grpc/server.rs (hunk 6 above)
Comment (amsmith-pro) [reply server/src/grpc/server.rs]: https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

File: server/src/grpc/server.rs (hunk 7 above)
Comment (amsmith-pro) [reply server/src/grpc/server.rs]: https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

File: server/src/grpc/server.rs (hunk 8 above)
Comment (amsmith-pro) [reply server/src/grpc/server.rs]: https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

File: server/src/grpc/server.rs (hunk 9 above)
Comment (amsmith-pro) [reply server/src/grpc/server.rs]: https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

File: server/src/grpc/server.rs (hunk 10 above)
Comment (amsmith-pro) [reply server/src/grpc/server.rs]: https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

File: server/src/grpc/server.rs (hunk 11 above)
Comment (amsmith-pro) [reply server/src/grpc/server.rs]: https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

File: server/src/grpc/server.rs (hunk 12 above)
Comment (amsmith-pro) [reply server/src/grpc/server.rs:360]: https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1

**Comment by amsmith-pro** [comment]
@owlot Updated log functions, added tests - 

Switched to the newer way of instantiating grpc clients so that it could be unit tested, works great!

Also changed the behavior of when timestamps can't be converted (which shouldn't happen, but jic) in `update_restrictions` to just set start/end time to None (start immediately/end never) rather than discarding the restriction, which could have some hairy real-life consequences

**Review by owlot — APPROVED** [review]

**Pull request merged**